### PR TITLE
Adds Ancient Spellbook and New Spell, "Poly" Morph (Work in Progress)

### PR DESCRIPTION
**Pull request by Arthurdentist**
Throwing this up now mostly for ideas to take this concept further. Its still a complete hack job that needs a lot of work.
## Ancient Spellbook
This PR adds a new single use spellbook to the game, the 'ancient spellbook'. It contains a random spell that can't be acquired under other normal circumstances.  So far the only possible spells are 'Disintegrate' and a new Spell, "Poly"Morph. 

To get the ancient spellbook you have to choose the spellbook bundle, and even then it isn't guaranteed you will get it. I intend to add at least one more unique spell to the spellbook; but I need to work on that. 

## "Poly"morph
This is a ranged spell that the wizard casts on a single target. The target is turned into a parrot for a short duration of time rendering them (mostly) harmless. Whilst they are a parrot they cannot be killed, so this spell isn't a way to get cheap and easy kills. Its effectively an alternative to the statue spell. 

It can remove an annoying person for a short amount of time but unlike the statue spell it can be cast at range. You can't use it to easily kill someone however, which you can do with the statue spell. The cooldown is pretty long and I don't *think* it is that powerful as a spell, but I am certainly up for tweaking it to make it properly balanced. 

I need to consider a better method of making an 'invincible' parrot. The method I am using currently is not at all pretty and I am sure there is a better way of doing it. 

## Things to do

- [ ] Code a third spell for the ancient spellbook
- [ ] Figure out a better way of making invincible parrots
- [ ] Currently when a person stops being a parrot, a random headset is dropped beneath their feet because new parrots spawn with a random headset. This ought to be fixed as well.
- [ ] Unique graphics for the ancient spellbook
- [ ] Balance the shit out of everything
- [ ] Fix any other dumb mistakes I have made

## Video
I'll make a video of the spell / book in action later today so you can see precisely how it all works. Just want to put this up first.

### CODE REVIEW

**Comment by Arthurdentist** [comment]
Already conflicted and im not sure why. Wew

Nevermind figured it out. Will address tomorrow.

File: code/modules/spells/targeted/parrotmorph.dm [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	hud_state = "wiz_parrotmorph"
+
+spell/targeted/parrotmorph/cast(var/list/targets)
+	..()
+	for(var/mob/living/target in targets)
+		target.flash_eyes(visual = 1)
+		var/mob/M = target.transmogrify(/mob/living/simple_animal/parrot/polymorph) //"invincible" parrot
+		M.name = target.name
+		spawn(2 MINUTES)
+		M.transmogrify()
```
Comment (unid15) [inline code/modules/spells/targeted/parrotmorph.dm:29]: Indentation after spawn()

**Comment by clusterfack** [comment]
I was right, we really are becoming runescape

**Comment by Arthurdentist** [comment]
Ice Barrage soon

but actually I couldn't think of a better name for it. If you got one im all ears.

File: code/modules/spells/targeted/parrotmorph.dm [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		to_chat(M, "<span class=sinister><B>The spell has worn off! You are no longer a parrot!</span></B>")
+
+
+
+
+/mob/living/simple_animal/parrot/polymorph //invincible parrot for polymorph spell. Turns victim into a parrot for a temporary amount of time. Don't want wizard to secure easy kills.
+	health = 99999999999 //got to be better ways of doing this but I haven't figured it out yet
+	maxHealth = 99999999999
```
Comment (ghost) [inline code/modules/spells/targeted/parrotmorph.dm:37]: Would probably be better to give em status_flag = CANSTUN|CANKNOCKDOWN|CANPARALYSE|CANPUSH|GODMODE

**Comment by PJB3005** [comment]
I've already had EI NATH of this PR.

File: code/game/gamemodes/wizard/spellbook.dm [hunk 3]
```diff
@@ -742,3 +742,21 @@
...
+	spell = null
+	icon_state = "book"
+	desc = "A book of lost and forgotten knowledge"
+	spellname = "forgotten knowledge"
+
+/obj/item/weapon/spellbook/oneuse/ancient/New()
+	..()
+	spell = pick(possible_spells)
+
+/obj/item/weapon/spellbook/oneuse/ancient/recoil(mob/living/carbon/user as mob)
```
Comment (PJB3005) [inline code/game/gamemodes/wizard/spellbook.dm:759]: Remove `as mob`

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/modules/spells/targeted/parrotmorph.dm [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+		M.name = target.name
+		spawn(2 MINUTES)
+		M.transmogrify()
+		to_chat(M, "<span class=sinister><B>The spell has worn off! You are no longer a parrot!</span></B>")
+
+
+
+
+/mob/living/simple_animal/parrot/polymorph //invincible parrot for polymorph spell. Turns victim into a parrot for a temporary amount of time. Don't want wizard to secure easy kills.
+	health = 99999999999 //got to be better ways of doing this but I haven't figured it out yet
```
Comment (ComicIronic) [inline code/modules/spells/targeted/parrotmorph.dm:36]: use `INFINITY`.

**Comment by Diavill** [comment]
Summon swords when

File: code/modules/spells/targeted/parrotmorph.dm (hunk 2 above)
Comment (Shadowmech88) [reply code/modules/spells/targeted/parrotmorph.dm:37]: Could also just override the procs for the parrot taking damage and/or dying and make them do nothing.

**Comment by Durgen** [comment]
How about a spell that would increase a target's susceptibility to damage for a certain amount of time?
Maybe the spell that could halve their brute resistance, burn resistance, etc.

**Comment by icantthinkofanameritenow** [comment]
I'd suggest one to turn someone into a frog, but the parrot thing kind of covers that already. 

Maybe a self-healing spell that acts like tricord and stops internal bleeding? I know there's no way in hell that it would be balanced as a stand-alone spell, but if it only appears in a rare spellbook which itself isn't guaranteed to contain that spell then it should be fine.

**Comment by Blithering** [comment]
What happens when you revert while being carried?

**Comment by Shadowmech88** [comment]
Un-transmogrifying in general places your old body on the turf of your current body, then moves you to your old body and deletes your current body.

You'd probably be fine, but I'm not sure what would happen to the animal holder if the animal inside it gets deleted.

**Comment by Blithering** [comment]
@Shadowmech88 That might be answered in https://github.com/vgstation-coders/vgstation13/issues/14689 
My big worry was wizards using the spell to nullspace the crew one parrot at a time.

**Comment by Arthurdentist** [comment]
I can't fix binary merge conflicts im not clever enough so lets take this from the top and pretend it never happened.

**Pull request closed without merging**